transport/src/main/java/io/netty/channel/EventLoopGroups.java

```diff
@@ -0,0 +1,179 @@
+     */
+    public static EventLoopGroup fastestNonBlocking() {
+        return fastestNonBlocking(0);
+    }
+
+    /**
+     * Returns fastest non-blocking {@link EventLoopGroup} instance for running platform. Tries to initialize native
+     * implementation for running platform (epoll on Linux for example) first and fallbacks to
+     * {@link io.netty.channel.nio.NioEventLoopGroup} if native implementation for running platform is not available.
+     *
+     * @param nThreads number of threads
+     * @return event loop group
+     */
+    public static EventLoopGroup fastestNonBlocking(int nThreads) {
+        return fastestNonBlocking(nThreads, null);
+    }
+
+    /**
+     * Returns fastest non-blocking {@link EventLoopGroup} instance for running platform. Tries to initialize native
+     * implementation for running platform (epoll on Linux for example) first and fallbacks to
+     * {@link io.netty.channel.nio.NioEventLoopGroup} if native implementation for running platform is not available.
+     *
+     * @param nThreads      number of threads
+     * @param threadFactory thread factory
+     * @return event loop group
+     */
+    public static EventLoopGroup fastestNonBlocking(int nThreads, ThreadFactory threadFactory) {
+        for (Constructor<? extends EventLoopGroup> constructor : EVENT_LOOP_CONSTRUCTORS) {
+            try {
+                return constructor.newInstance(nThreads, threadFactory);
+            } catch (Exception e) {
+                logger.debug("Cannot invoke event loop group constructor: {}", constructor, e);
+            }
+        }
+        throw new IllegalStateException(
+                "Cannot successfully invoke any of discovered createEventLoopGroupConstructors: " +
+                        EVENT_LOOP_CONSTRUCTORS);
+    }
+
+    /**
+     * Tries to load {@link EventLoopGroup} class.
+     *
+     * @param fqcn event loop group fully qualified class name.
+     * @return event loop group class on success, otherwise <code>null</code>.
+     * @see #loadClass(String, Class)
+     */
+    protected static Class<? extends EventLoopGroup> loadEventLoopGroupClass(String fqcn) {
+        return loadClass(fqcn, EventLoopGroup.class);
+    }
+
+    /**
+     * Tries to load class with specified fully qualified class name.
+     *
+     * @param fqcn      fully qualified class name.
+     * @param classType class into which loaded class should be cast.
+     * @param <T>       class type.
+     * @return class on success, otherwise <code>null</code>.
+     */
+    protected static <T> Class<T> loadClass(String fqcn, Class<T> classType) {
+        try {
+            Class<?> clazz = Class.forName(fqcn);
+            return (Class<T>) clazz.asSubclass(classType);
+        } catch (ClassNotFoundException e) {
+            logger.trace("Unable to load class {}: class is not present on classpath.", fqcn);
+            return null;
+        }
+    }
+
+    protected static String ucFirst(String s) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this "protected" modifier. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS2156)